src/DataAccess/DoctrineApplicationRepository.php

```diff
@@ -55,12 +56,23 @@ public function storeApplication( Application $application ): void {
 	private function insertApplication( Application $application ): void {
 		$doctrineApplication = new DoctrineApplication();
 		$this->updateDoctrineApplication( $doctrineApplication, $application );
-
+		$doctrineApplication->setAddressChange(
+			new AddressChange( $this->getAddressChangeType( $application->getApplicant() ) )
+		);
 		$this->table->persistApplication( $doctrineApplication );
 
 		$application->assignId( $doctrineApplication->getId() );
 	}
 
+	private function getAddressChangeType( Applicant $applicant ): string {
+		if ( $applicant->getName()->isPrivatePerson() ) {
```

Comment:
`$applicant->isPrivatePerson()` might be better

https://en.wikipedia.org/wiki/Law_of_Demeter

There also is the notion in DDD that you should only talk to an Aggregate (in this case the membership application) directly and not its children. This is especially important when giving commands (which is not happening here).